include/tlb.h

```diff
@@ -26,9 +16,24 @@ typedef uint32_t pte_t;
 #define SET_V(e,b)      (e) = ((e) & ~(V_MASK)) | (((b) << 1) & V_MASK)
 #define SET_G(e,g)      (e) = ((e) & ~(G_MASK)) | (((g)     ) & G_MASK)
 
-void tlb_init();
-void tlb_set_page_table(pmap_t *pmap);
-void tlb_print();
+#define PTE_BASE  MIPS_KSEG2_START
+
+void     tlb_init();
+void     tlb_print();
+
+/* Codescape GNU tlb functions don't preserve current asid, these functions are wrappers 
+ * which preserve asid. For documentation please see 13.2 chapter of Codescape GNU Tool for mips */
+
+unsigned tlb_size();
+void     tlb_inval(tlbhi_t hi);
```

Comment:
Make those function names self-explanatory. If I hadn't knew MIPS I would have not a slightest idea what they do. Well... in fact `tlb_rwr2` is a mystery to me.